--- a/corelib/parsing_library_libconfig.c
+++ b/corelib/parsing_library_libconfig.c
@@ -42,22 +42,29 @@
 static void get_value_libconfig(const config_setting_t *e, void *dest, field_type_t expected_type)
 {
 	int type = config_setting_type(e);
-	if (type != map_field_type(expected_type))
-		return;
-	switch (type) {
-	case CONFIG_TYPE_INT:
+	if (type != map_field_type(expected_type)) {
+		/* Weaken type equality requirements for INT/INT64 */
+		if ((type == CONFIG_TYPE_INT && expected_type == TYPE_INT64) ||
+		    (type == CONFIG_TYPE_INT64 && expected_type == TYPE_INT)) {
+			/* ignore type mismatch, handled well by libconfig */
+		} else
+			return;
+	}
+
+	switch (expected_type) {
+	case TYPE_INT:
 		*(int *)dest = config_setting_get_int(e);
 		break;
-	case CONFIG_TYPE_INT64:
+	case TYPE_INT64:
 		*(long long *)dest = config_setting_get_int64(e);
 		break;
-	case CONFIG_TYPE_STRING:
+	case TYPE_STRING:
 		dest = (void *)config_setting_get_string(e);
 		break;
-	case CONFIG_TYPE_BOOL:
+	case TYPE_BOOL:
 		*(bool *)dest = config_setting_get_bool(e);
 		break;
-	case CONFIG_TYPE_FLOAT:
+	case TYPE_FLOAT:
 		*(double *)dest = config_setting_get_float(e);
 		break;
 		/* Do nothing, add if needed */
