--- a/tools/depmod.c
+++ b/tools/depmod.c
@@ -2612,8 +2612,13 @@
 			int mode = 0644;
 			int fd;
 
+#ifdef __USE_TIME_BITS64
+			snprintf(tmp, sizeof(tmp), "%s.%i.%lli.%lli", itr->name, getpid(),
+					tv.tv_usec, tv.tv_sec);
+#else
 			snprintf(tmp, sizeof(tmp), "%s.%i.%li.%li", itr->name, getpid(),
 					tv.tv_usec, tv.tv_sec);
+#endif
 			fd = openat(dfd, tmp, flags, mode);
 			if (fd < 0) {
 				ERR("openat(%s, %s, %o, %o): %m\n",
