--- a/meson.build	2022-03-24 09:43:06.000000000 
+++ b/meson.build	2022-03-24 09:29:57.000000000 
@@ -1360,7 +1360,7 @@
 conf.set10('HAVE_GNUTLS', libgnutls.found())
 
 libopenssl = dependency('openssl',
-                        version : '>= 1.1.0',
+                        version : '>= 1.0.0',
                         required : get_option('openssl'))
 conf.set10('HAVE_OPENSSL', libopenssl.found())
 
--- b/src/shared/openssl-util.h	2022-01-18 07:35:43.000000000 
+++ b/src/shared/openssl-util.h	2022-03-24 09:50:07.000000000 
@@ -39,6 +39,27 @@
 #    include <openssl/store.h>
 #  endif
 
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+static inline void *OPENSSL_zalloc(size_t num)
+{
+	void *ret = OPENSSL_malloc(num);
+	if (ret)
+		memset(ret, 0, num);
+	return ret;
+}
+
+static inline EVP_MD_CTX *EVP_MD_CTX_new(void)
+{
+	return OPENSSL_zalloc(sizeof(EVP_MD_CTX));
+}
+
+static inline void EVP_MD_CTX_free(EVP_MD_CTX *ctx)
+{
+	EVP_MD_CTX_cleanup(ctx);
+	OPENSSL_free(ctx);
+}
+#endif
+
 DEFINE_TRIVIAL_CLEANUP_FUNC_FULL_MACRO(void*, OPENSSL_free, NULL);
 DEFINE_TRIVIAL_CLEANUP_FUNC_FULL(X509_NAME*, X509_NAME_free, NULL);
 DEFINE_TRIVIAL_CLEANUP_FUNC_FULL(EVP_PKEY_CTX*, EVP_PKEY_CTX_free, NULL);
