--- a/src/basic/macro.h
+++ b/src/basic/macro.h
@@ -379,12 +379,21 @@
 
 /* Iterate through each argument passed. All must be the same type as 'entry' or must be implicitly
  * convertible. The iteration variable 'entry' must already be defined. */
+#if __GNUC__ < 8
+#define FOREACH_ARGUMENT(entry, ...)                                     \
+        _FOREACH_ARGUMENT(entry, UNIQ_T(_entries_, UNIQ), UNIQ_T(_current_, UNIQ), ##__VA_ARGS__)
+#define _FOREACH_ARGUMENT(entry, _entries_, _current_, ...)         \
+        for (typeof(entry) _entries_[] = { __VA_ARGS__ }, *_current_ = _entries_; \
+             ((long)(_current_ - _entries_) < (long)ELEMENTSOF(_entries_)) && ({ entry = *_current_; true; }); \
+             _current_++)
+#else
 #define FOREACH_ARGUMENT(entry, ...)                                     \
         _FOREACH_ARGUMENT(entry, UNIQ_T(_entries_, UNIQ), UNIQ_T(_current_, UNIQ), UNIQ_T(_va_sentinel_, UNIQ), ##__VA_ARGS__)
 #define _FOREACH_ARGUMENT(entry, _entries_, _current_, _va_sentinel_, ...)      \
         for (typeof(entry) _va_sentinel_[1] = {}, _entries_[] = { __VA_ARGS__ __VA_OPT__(,) _va_sentinel_[0] }, *_current_ = _entries_; \
              ((long)(_current_ - _entries_) < (long)(ELEMENTSOF(_entries_) - 1)) && ({ entry = *_current_; true; }); \
              _current_++)
+#endif
 
 #define DECIMAL_STR_FMT(x) _Generic((x),        \
         char: "%c",                             \
--- a/src/shared/varlink-idl.h
+++ b/src/shared/varlink-idl.h
@@ -103,6 +103,51 @@
 #define VARLINK_DEFINE_ENUM_VALUE(_name) \
         { .name = #_name, .field_type = VARLINK_ENUM_VALUE }
 
+#if __GNUC__ < 8
+#define VARLINK_DEFINE_METHOD_NAME(_name, ...)                          \
+        const VarlinkSymbol vl_method_ ## _name = {                     \
+                .name = #_name,                                         \
+                .symbol_type = VARLINK_METHOD,                          \
+                .fields = { __VA_ARGS__ },                              \
+        }
+#define VARLINK_DEFINE_METHOD(_name, ...)                               \
+        VARLINK_DEFINE_METHOD_NAME(_name, ##__VA_ARGS__, {})
+
+#define VARLINK_DEFINE_ERROR_NAME(_name, ...)                           \
+        const VarlinkSymbol vl_error_ ## _name = {                      \
+                .name = #_name,                                         \
+                .symbol_type = VARLINK_ERROR,                           \
+                .fields = { __VA_ARGS__ },                              \
+        }
+#define VARLINK_DEFINE_ERROR(_name, ...)                                \
+        VARLINK_DEFINE_ERROR_NAME(_name, ##__VA_ARGS__, {})
+
+#define VARLINK_DEFINE_STRUCT_TYPE_NAME(_name, ...)                     \
+        const VarlinkSymbol vl_type_ ## _name = {                       \
+                .name = #_name,                                         \
+                .symbol_type = VARLINK_STRUCT_TYPE,                     \
+                .fields = { __VA_ARGS__ },                              \
+        }
+#define VARLINK_DEFINE_STRUCT_TYPE(_name, ...)                          \
+        VARLINK_DEFINE_STRUCT_TYPE_NAME(_name, ##__VA_ARGS__, {})
+
+#define VARLINK_DEFINE_ENUM_TYPE_NAME(_name, ...)                       \
+        const VarlinkSymbol vl_type_ ## _name = {                       \
+                .name = #_name,                                         \
+                .symbol_type = VARLINK_ENUM_TYPE,                       \
+                .fields = { __VA_ARGS__ },                              \
+        }
+#define VARLINK_DEFINE_ENUM_TYPE(_name, ...)                            \
+        VARLINK_DEFINE_ENUM_TYPE_NAME(_name, ##__VA_ARGS__, {})
+
+#define VARLINK_DEFINE_INTERFACE_NAME(_name, _full_name, ...)           \
+        const VarlinkInterface vl_interface_ ## _name = {               \
+                .name = (_full_name),                                   \
+                .symbols = { __VA_ARGS__ },           \
+        }
+#define VARLINK_DEFINE_INTERFACE(_name, _full_name, ...)                \
+        VARLINK_DEFINE_INTERFACE_NAME(_name, _full_name, ##__VA_ARGS__, NULL)
+#else
 #define VARLINK_DEFINE_METHOD(_name, ...)                               \
         const VarlinkSymbol vl_method_ ## _name = {                     \
                 .name = #_name,                                         \
@@ -136,6 +171,7 @@
                 .name = (_full_name),                                   \
                 .symbols = { __VA_ARGS__ __VA_OPT__(,) NULL},           \
         }
+#endif
 
 int varlink_idl_dump(FILE *f, int use_colors, const VarlinkInterface *interface);
 int varlink_idl_format(const VarlinkInterface *interface, char **ret);
