--- a/Modules/_ssl.c	2024-09-13 14:46:13.000000000 
+++ b/Modules/_ssl.c	2024-09-13 15:18:16.000000000 
@@ -4689,7 +4689,7 @@
     return 0;
 }
 
-#if OPENSSL_VERSION_NUMBER < 0x30300000L
+#if (OPENSSL_VERSION_NUMBER > 0x10100000L) && (OPENSSL_VERSION_NUMBER < 0x30300000L)
 static X509_OBJECT *x509_object_dup(const X509_OBJECT *obj)
 {
     int ok;
@@ -4762,7 +4762,11 @@
     int x509 = 0, crl = 0, ca = 0, i;
 
     store = SSL_CTX_get_cert_store(self->ctx);
+#if OPENSSL_VERSION_NUMBER > 0x10100000L
     objs = X509_STORE_get1_objects(store);
+#else
+    objs = X509_STORE_get0_objects(store);
+#endif
     if (objs == NULL) {
         PyErr_SetString(PyExc_MemoryError, "failed to query cert store");
         return NULL;
@@ -4785,7 +4789,9 @@
                 break;
         }
     }
+#if OPENSSL_VERSION_NUMBER > 0x10100000L
     sk_X509_OBJECT_pop_free(objs, X509_OBJECT_free);
+#endif
     return Py_BuildValue("{sisisi}", "x509", x509, "crl", crl,
         "x509_ca", ca);
 }
@@ -4817,7 +4823,11 @@
     }
 
     store = SSL_CTX_get_cert_store(self->ctx);
+#if OPENSSL_VERSION_NUMBER > 0x10100000L
     objs = X509_STORE_get1_objects(store);
+#else
+    objs = X509_STORE_get0_objects(store);
+#endif
     if (objs == NULL) {
         PyErr_SetString(PyExc_MemoryError, "failed to query cert store");
         goto error;
@@ -4850,11 +4860,15 @@
         }
         Py_CLEAR(ci);
     }
+#if OPENSSL_VERSION_NUMBER > 0x10100000L
     sk_X509_OBJECT_pop_free(objs, X509_OBJECT_free);
+#endif
     return rlist;
 
   error:
+#if OPENSSL_VERSION_NUMBER > 0x10100000L
     sk_X509_OBJECT_pop_free(objs, X509_OBJECT_free);
+#endif
     Py_XDECREF(ci);
     Py_XDECREF(rlist);
     return NULL;
