--- a/configure.ac	2024-09-13 13:28:50.000000000 
+++ b/configure.ac	2024-09-13 00:59:48.000000000 
@@ -6841,13 +6841,12 @@
     AC_LINK_IFELSE([AC_LANG_PROGRAM([
       #include <openssl/opensslv.h>
       #include <openssl/ssl.h>
-      #if OPENSSL_VERSION_NUMBER < 0x10101000L
-        #error "OpenSSL >= 1.1.1 is required"
-      #endif
       static void keylog_cb(const SSL *ssl, const char *line) {}
     ], [
+      #if OPENSSL_VERSION_NUMBER < 0x10101000L
+      #define TLS_client_method SSLv23_client_method 
+      #endif
       SSL_CTX *ctx = SSL_CTX_new(TLS_client_method());
-      SSL_CTX_set_keylog_callback(ctx, keylog_cb);
       SSL *ssl = SSL_new(ctx);
       X509_VERIFY_PARAM *param = SSL_get0_param(ssl);
       X509_VERIFY_PARAM_set1_host(param, "python.org", 0);
@@ -6866,15 +6865,9 @@
     AC_LINK_IFELSE([AC_LANG_PROGRAM([
       #include <openssl/opensslv.h>
       #include <openssl/evp.h>
-      #if OPENSSL_VERSION_NUMBER < 0x10101000L
-        #error "OpenSSL >= 1.1.1 is required"
-      #endif
     ], [
       OBJ_nid2sn(NID_md5);
       OBJ_nid2sn(NID_sha1);
-      OBJ_nid2sn(NID_sha3_512);
-      OBJ_nid2sn(NID_blake2b512);
-      EVP_PBE_scrypt(NULL, 0, NULL, 0, 2, 8, 1, 0, NULL, 0);
     ])], [ac_cv_working_openssl_hashlib=yes], [ac_cv_working_openssl_hashlib=no])
   ])
 ])
--- a/Modules/_ssl.c	2024-09-13 14:46:13.000000000 
+++ b/Modules/_ssl.c	2024-09-13 15:18:16.000000000 
@@ -2108,7 +2108,9 @@
 /*[clinic end generated code: output=3d174ead2e42c4fd input=0bfe149da8fe6306]*/
 {
     STACK_OF(SSL_CIPHER) *server_ciphers;
+#if OPENSSL_VERSION_NUMBER > 0x10100000L
     STACK_OF(SSL_CIPHER) *client_ciphers;
+#endif
     int i, len;
     PyObject *res;
     const SSL_CIPHER* cipher;
@@ -2124,9 +2126,11 @@
     server_ciphers = SSL_get_ciphers(self->ssl);
     if (!server_ciphers)
         Py_RETURN_NONE;
+#if OPENSSL_VERSION_NUMBER > 0x10100000L
     client_ciphers = SSL_get_client_ciphers(self->ssl);
     if (!client_ciphers)
         Py_RETURN_NONE;
+#endif
 
     res = PyList_New(sk_SSL_CIPHER_num(server_ciphers));
     if (!res)
@@ -2134,8 +2138,10 @@
     len = 0;
     for (i = 0; i < sk_SSL_CIPHER_num(server_ciphers); i++) {
         cipher = sk_SSL_CIPHER_value(server_ciphers, i);
+#if OPENSSL_VERSION_NUMBER > 0x10100000L
         if (sk_SSL_CIPHER_find(client_ciphers, cipher) < 0)
             continue;
+#endif
 
         PyObject *tup = cipher_to_tuple(cipher);
         if (!tup) {
@@ -4694,7 +4700,7 @@
     return 0;
 }
 
-#if OPENSSL_VERSION_NUMBER < 0x30300000L
+#if (OPENSSL_VERSION_NUMBER > 0x10100000L) && (OPENSSL_VERSION_NUMBER < 0x30300000L)
 static X509_OBJECT *x509_object_dup(const X509_OBJECT *obj)
 {
     int ok;
@@ -4767,7 +4773,11 @@
     int x509 = 0, crl = 0, ca = 0, i;
 
     store = SSL_CTX_get_cert_store(self->ctx);
+#if OPENSSL_VERSION_NUMBER > 0x10100000L
     objs = X509_STORE_get1_objects(store);
+#else
+    objs = X509_STORE_get0_objects(store);
+#endif
     if (objs == NULL) {
         PyErr_SetString(PyExc_MemoryError, "failed to query cert store");
         return NULL;
@@ -4790,7 +4800,9 @@
                 break;
         }
     }
+#if OPENSSL_VERSION_NUMBER > 0x10100000L
     sk_X509_OBJECT_pop_free(objs, X509_OBJECT_free);
+#endif
     return Py_BuildValue("{sisisi}", "x509", x509, "crl", crl,
         "x509_ca", ca);
 }
@@ -4822,7 +4834,11 @@
     }
 
     store = SSL_CTX_get_cert_store(self->ctx);
+#if OPENSSL_VERSION_NUMBER > 0x10100000L
     objs = X509_STORE_get1_objects(store);
+#else
+    objs = X509_STORE_get0_objects(store);
+#endif
     if (objs == NULL) {
         PyErr_SetString(PyExc_MemoryError, "failed to query cert store");
         goto error;
@@ -4855,11 +4871,15 @@
         }
         Py_CLEAR(ci);
     }
+#if OPENSSL_VERSION_NUMBER > 0x10100000L
     sk_X509_OBJECT_pop_free(objs, X509_OBJECT_free);
+#endif
     return rlist;
 
   error:
+#if OPENSSL_VERSION_NUMBER > 0x10100000L
     sk_X509_OBJECT_pop_free(objs, X509_OBJECT_free);
+#endif
     Py_XDECREF(ci);
     Py_XDECREF(rlist);
     return NULL;
